nav2_route/include/nav2_route/types.hpp

```diff
@@ -184,6 +184,13 @@ struct Node
   Operations operations;     // Operations to perform related to the node
   SearchState search_state;  // State maintained by route search algorithm
 
+  void addEdge(
+    EdgeCost & cost, NodePtr node, unsigned int edgeid, Metadata & Metadata,
+    Operations & Operations)
+  {
+    neighbors.push_back({edgeid, this, node, cost, Metadata, Operations});
+  }
+
   void addEdge(EdgeCost & cost, NodePtr node, unsigned int edgeid)
```

Comment:
You can remove this implementation if you make your other one have default values for the metadata / operations arguments (eg `{}`) if not provided. Then both ways would work.